packages/alfa-dom/src/common/predicate.ts

```diff
@@ -0,0 +1,32 @@
+import { Predicate } from "@siteimprove/alfa-predicate";
+import {Element, Namespace, Node} from "..";
+
+const { and, equals, property } = Predicate;
+
+export function hasNamespace(
+  predicate: Predicate<Namespace> = () => true
+): Predicate<Element> {
+  return (element) => element.namespace.some(predicate);
+}
```

Comment:
This would be well-suited as `Element#hasNamespace(predicate: Predicate<Namespace>): boolean` and `Element.hasNamespace(predicate: Predicate<Namespace>): Predicate<Element>`.